test/test_pipeline_with_ddp.py

```diff
@@ -0,0 +1,177 @@
+# (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+import unittest
+
+import torch
+import torch.distributed as dist
+import torch.nn as nn
+from pippy.ManualPipelineStage import (
+    create_metadata_tensor,
+    extract_metadata_from_tensor,
+    get_stage_shapes,
+    ManualPipelineStage,
+    validate_stage_shapes,
+)
+from pippy.PipelineSchedule import PipelineScheduleGPipe
+from torch.distributed.device_mesh import init_device_mesh
+from torch.nn.modules.loss import F
+
+# torch.testing._internal.common_distributed requies "expecttest"
+from torch.testing._internal.common_distributed import MultiProcessTestCase
+from torch.testing._internal.common_utils import FILE_SCHEMA
+
+
+class DDPAROnce(torch.nn.Module):
```

Comment:
How representative is this helper?
Is there a preference between using this helper vs using `with ddp_model.no_sync()`?